blocks/embed/embed.js

```diff
@@ -0,0 +1,255 @@
+  }
+};
+
+/**
+ * Gets consent status for a given source.
+ * @function getConsent
+ * @memberof EmbedBlock
+ * @param {string} source - The source to check consent for.
+ * @param {string} [key='videoConsent'] - The key to retrieve consent from in sessionStorage.
+ * @returns {boolean} - The consent status for the given source.
+ */
+const getConsent = (source, key = 'videoConsent') => {
+  let output = false;
+  if (sessionStorage.getItem(key)) {
+    const consentSet = new Set(sessionStorage.getItem(key).split(','));
+    if (consentSet.has(source)) {
+      output = true;
+    }
+  }
+  return output;
+};
+
+/**
+ * Embeds a YouTube video.
+ * @function embedYoutube
+ * @memberof EmbedBlock
+ * @param {URL} url - The URL of the YouTube video.
+ * @param {boolean} [autoplay=true] - Whether to autoplay the video. Defaults to true.
+ * @returns {string} - The HTML code for embedding the YouTube video.
+ */
+const embedYoutube = (url, autoplay = true) => {
+  const usp = new URLSearchParams(url.search);
+  const suffix = autoplay ? '&muted=1&autoplay=1' : '';
+  let vid = usp.get('v') ? encodeURIComponent(usp.get('v')) : '';
+  const embed = url.pathname;
+  if (url.origin.includes('youtu.be')) {
+    [, vid] = url.pathname.split('/');
+  }
+
+  vid = decodeURI(`${vid}`).replace(/\\/g, '');
+
+  const embedHTML = `<div class="video-embed-container">
+      <iframe src="https://www.youtube.com${vid ? `/embed/${vid}?rel=0&v=${vid}${suffix}` : embed}"
+      class="video-embed-iframe"
+      allow="autoplay; fullscreen; picture-in-picture; encrypted-media; accelerometer; gyroscope; picture-in-picture"
+      allowfullscreen=""
+      scrolling="no"
+      title="Content from Youtube"
+      loading="lazy"></iframe>
+      </div>`;
+  return embedHTML;
+};
+
+/**
+ * Embeds a VideoJS Video.
+ * @function embedVideoJS
+ * @memberof EmbedBlock
+ * @param {URL} url - The URL of the VideoJS video.
+ * @param {boolean} autoplay - Whether autoplay is enabled or not.
+ * @returns {string} - The HTML for embedding the VideoJS video.
+ * @todo Validate whether this src URL pattern is really correct for all DGL videos (contact Pete Chamberlin)
+ * @todo Determine whether this is the best place to include the css and js lib file for videoJS
+ * @todo CSS styling pass needed to ensure the VideoJS player is brand compliant
+ */
+const embedVideoJS = (url, autoplay = true, poster = null) => {
+  // eslint-disable-next-line comma-dangle
+  const [, , vid,] = url.pathname.split('/');
+  const embedHTML = `
+    <link rel="stylesheet" href="/styles/video-js.css">
+    <script src="/scripts/video-js.lib.js"></script>
+    <div class="embed-container embed-no-padding">
+      <video
+        id="video-js"
+        class="video-js embed-iframe"
+        controls ${autoplay ? 'autoplay' : ''}
+        preload="auto"
+        poster="${poster}"
+        data-setup='{}'>
+        <source src="https://d.dam.sap.com/m/${vid}/hls.m3u8" type="video/mp4"></source>
```

Comment:
Is it always the same source host?

Reply:
As far as I know, yes.  Calling d.dam results in a redirect to v.dam.  I actually have a call scheduled with the digital library team this week to get a better understanding of how that works and to talk through the CORS issue.